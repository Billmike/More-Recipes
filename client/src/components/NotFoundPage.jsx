import React from 'react';
import { Link } from 'react-router-dom';
import Emoji from 'react-emoji-render';
import Footer from './Footer';

const NotFoundPage = () => (
  <div>
    <div className="section-bck">
      <p className="p-section">
        You've gone too far in search of a dessert, init? <Emoji text=":'(" />{' '}
        <Link to="/"> Go back</Link>
      </p>
    </div>
    <Footer />
  </div>
);

export default NotFoundPage;
