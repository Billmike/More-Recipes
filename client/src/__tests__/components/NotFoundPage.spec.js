import React from 'react';
import { shallow } from 'enzyme';
import Emoji from 'react-emoji-render';
import { Link } from 'react-router-dom';
import NotFoundPage from '../../components/NotFoundPage';

test('Should render the 404 page correctly', () => {
  const wrapper = shallow(<NotFoundPage />);
  expect(wrapper.find('div').length).toBe(2);
  expect(wrapper.find('p').text()).toBe(
    "You've gone too far in search of a dessert, init?<Emoji /> <Link />"
  );
  expect(wrapper).toMatchSnapshot();
});
