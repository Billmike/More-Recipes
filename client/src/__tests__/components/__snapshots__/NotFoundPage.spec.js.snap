// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the 404 page correctly 1`] = `
<div>
  <div
    className="section-bck"
  >
    <p
      className="p-section"
    >
      You've gone too far in search of a dessert, init? 
      <Emoji
        text=":'("
      />
       
      <Link
        replace={false}
        to="/"
      >
         Go back
      </Link>
    </p>
  </div>
  <Footer />
</div>
`;
